python/adelphi/tests/integration/__init__.py

```diff
@@ -0,0 +1,192 @@
+import glob
+import logging
+import os
+import shutil
+import sys
+import tempfile
+import time
+
+if os.name == 'posix' and sys.version_info[0] < 3:
+    import subprocess32 as subprocess
+else:
+    import subprocess
+
+from collections import namedtuple
+
+from cassandra.cluster import Cluster
+
+import docker
+
+from tests.util.schemadiff import cqlDigestGenerator
+
+# Default C* versions to include in all integration tests
+CASSANDRA_VERSIONS = ["2.1.22", "2.2.19", "3.0.23", "3.11.9", "4.0-beta3"]
+
+logging.basicConfig(filename="adelphi.log", level=logging.INFO)
+log = logging.getLogger('adelphi')
+
+class DockerSchemaTestMixin:
+
+    def connectToLocalCassandra(self):
+        session = None
+        while not session:
+            try:
+                cluster = Cluster(["127.0.0.1"],port=9042)
+                session = cluster.connect()
+
+                # Confirm that the session is actually functioning before calling things good
+                rs = session.execute("select * from system.local")
+                log.info("Connected to Cassandra cluster, first row of system.local: {}".format(rs.one()))
+                log.info("Cassandra cluster ready")
+                return (cluster,session)
+            except:
+                log.info("Couldn't quite connect yet, will retry")
+                time.sleep(1)
+
+
+    def createSchema(self, session=None):
+        schemaPath = self.getBaseSchemaPath()
+        log.info("Creating schema on Cassandra cluster from file {}".format(schemaPath))
+        with open(schemaPath) as schema:
+            buff = ""
+            for line in schema:
+                realLine = line.strip()
+                if len(realLine) == 0:
+                    log.debug("Skipping empty statement")
+                    continue
+                if realLine.startswith("//"):
+                    log.debug("Skipping commented statement {}".format(stmt))
+                    continue
+                buff += (" " if len(buff) > 0 else "")
+                buff += realLine
+                if realLine.endswith(';'):
+                    log.debug("Executing statement {}".format(buff))
+                    try:
+                        session.execute(buff)
+                    except:
+                        log.info("Exception executing statement: {}".format(buff), exc_info=sys.exc_info()[0])
+                    buff = ""
```

Comment:
After recent changes this fn does now have some fairly strong resemblance to schemadiff.cqlDigestGenerator (see https://github.com/datastax/adelphi/pull/104/files#diff-2bc5a69b0a6a26da146863eadfe81e272b9cbf29ad2ee38992262a0d10ed8058R8-R23) but I think it still is worthwhile to keep this guy separate.  Doing so allows us to emphasize the role of cqlDigestGenerator() as a generator fn while thinking of this guy as a basic fn.

We certainly _could_ create a higher-level wrapper function which would contain the line-for-line reading from a file object.  We'd then have to supply another fn defining what do with each accumulated statement (i.e. the contents of a buffer once we find a trailng ';').  But note that these fns have different logic around when to ignore a line... so that would have to be abstracted too.  And at that point it seemed like it was getting complicated enough that just explicitly re-using the logic here was the cleanest bet.